src/fsharp/TastOps.fs

```diff
@@ -3231,6 +3236,16 @@ let (|TypeDefOfExpr|_|) g expr =
     | Expr.App (Expr.Val (vref, _, _), _, [ty], [], _) when isTypeDefOfValRef g vref -> Some ty
     | _ -> None
 
+let (|NameOfExpr|_|) g expr = 
+    match expr with 
+    | Expr.App(Expr.Val(vref,_,_),_,[ty],[],_) when isNameOfValRef g vref  -> Some ty
+    | _ -> None
+
+let (|SeqExpr|_|) g expr = 
```

Comment:
Why is this being added?

Reply:
It is cleanup of this line in `TypeChecker.fs`
```
                 | ApplicableExpr(_, Expr.Op (TOp.Coerce, _, [Expr.App (Expr.Val (vf, _, _), _, _, _, _)], _), _) when valRefEq cenv.g vf cenv.g.seq_vref -> true 
```